app/lib/smaxtec_api.rb

```diff
@@ -5,21 +5,46 @@ class SmaxtecApi
   SMAXTEC_API_PASSWORD = Rails.application.secrets.smaxtec_api_password
   SMAXTEC_API_BASE_URL = 'https://api-staging.smaxtec.com/api/v1'
 
-  def get_temperature
+  def get_sensor_readings
+    #define animal_id (smaxtec api identifier) for cows: topic_id => animal_id
+    animal_id = { 1 => '5722099ea80a5f54c631513d' }
+
+    # Assuming 1 report = 1 cow, otherwise get just the specified reports
+    Report.all.each do |report|
+      if animal_id[report.id]
+        sensors = Sensor.where(report_id: report.id, smaxtec_sensor: true)
+        if sensors
+          sensors.each do |sensor|
+            # temperature sensor
+            if sensor.sensor_type_id == 1
+              # get temperature data from smaxtec api
+              temperature = get_temperature(animal_id[report.id])
```

Comment:
I assume that a report is associated with a cow (e.g. 'Report Kuh Berta'). In the Smaxtec API the cow is identified by an animal_id. So at the beginning I define the animal_id hash array(?) and when iterating over every report I want to query the Smaxtec API with the associated animal_id to the report. But I only do this if the report has actually an animal_id (cow) associated with it (see line 14), because I don't know if its always the case.